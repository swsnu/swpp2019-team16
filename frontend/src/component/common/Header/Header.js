import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import PropTypes from 'prop-types';

HeaderMaterial.propTypes = {
  auth: PropTypes.bool,
  onClickLogo: PropTypes.func,
  onClickLogout: PropTypes.func,
  onClickPoint: PropTypes.func,
};

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
}));

export default function HeaderMaterial({
  auth,
  onClickLogo,
  onClickLogout,
  onClickChargePoint,
}) {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);
  const open = Boolean(anchorEl);

  const onClickMenu = event => {
    setAnchorEl(event.currentTarget);
  };

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            className={classes.title}
            onClick={onClickLogo}
          >
            Ya-Ta!
          </Typography>
          {auth && (
            <div>
              <IconButton
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={onClickMenu}
                color="inherit"
              >
                <AccountCircle />
              </IconButton>
              <Menu
                id="menu-appbar"
                anchorEl={anchorEl}
                anchorOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                keepMounted
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                open={open}
              >
                <MenuItem onClick={onClickChargePoint}>Charge</MenuItem>
                <MenuItem onClick={onClickLogout}>Log out</MenuItem>
              </Menu>
            </div>
          )}
        </Toolbar>
      </AppBar>
    </div>
  );
}
