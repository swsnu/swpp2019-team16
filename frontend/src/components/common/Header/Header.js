import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import PropTypes from 'prop-types';

export const HEADER_HEIGHT = '64px';

HeaderMaterial.propTypes = {
  user: PropTypes.object,
  onClickLogo: PropTypes.func,
  onCLickLogin: PropTypes.func,
  onClickLogout: PropTypes.func,
  onClickPoint: PropTypes.func,
};

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.primary.contrastText,
    zIndex: theme.zIndex.high,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  button: {
    color: 'white',
    cursor: 'pointer',
    paddingLeft: '1rem',
    paddingRight: '1rem',
    paddingBottom: '0.3rem',
    paddingTop: '0.3rem',
    marginBottom: '10px',
    marginTop: '10px',
  },
}));

export default function HeaderMaterial({
  user,
  auth,
  onClickLogo,
  onClickLogin,
  onClickLogout,
  onClickChargePoint,
}) {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);

  const open = Boolean(anchorEl);
  const onClickClose = () => {
    setAnchorEl(null);
  };

  const onClickMenu = event => {
    setAnchorEl(event.currentTarget);
  };

  return (
    <div className={classes.root}>
      <Toolbar>
        <Typography
          variant="h4"
          className={classes.title}
          onClick={onClickLogo}
        >
          Ya-Ta!
        </Typography>
        {user !== null && auth !== null ? (
          <div>
            <IconButton
              aria-label="account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={onClickMenu}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
            <Menu
              id="menu-appbar"
              anchorEl={anchorEl}
              anchorOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              keepMounted
              transformOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              open={open}
              onClose={onClickClose}
            >
              <MenuItem onClick={onClickChargePoint}>Charge point</MenuItem>
              <MenuItem onClick={onClickLogout}>Log out</MenuItem>
            </Menu>
          </div>
        ) : (
          <div
            className={classes.button}
            children={'LOGIN'}
            onClick={onClickLogin}
          />
        )}
      </Toolbar>
    </div>
  );
}
