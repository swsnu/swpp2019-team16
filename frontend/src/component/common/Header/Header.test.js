import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Header from './Header';

describe('<Header />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
