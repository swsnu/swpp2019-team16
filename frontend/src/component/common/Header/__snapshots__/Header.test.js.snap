// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Ya-Ta!
        </h6>
      </div>
    </header>
  </div>
</div>
`;
